hie-plugin-api/Haskell/Ide/Engine/PluginDescriptor.hs

```diff
@@ -214,7 +233,8 @@ data ParamVal (t :: ParamType) where
 class ValidResponse a where
   jsWrite :: a -> Object -- ^ Serialize to JSON Object
   jsRead  :: Object -> Parser a -- ^ Read from JSON Object
-
+  responseTypeName :: a -> T.Text -- ^ Type name in command description.
```

Comment:
I think what you are describing as a return type of a command is what I have been calling a semantic type.

From that perspective it makes sense to return something that identifies the semantic type that is returned, but to me it should be a property of hie, not just a string that gets defined ad hoc on a command.

For a semantic type it would make sense to distinguish between a list of files that may have been changed and a list of files that were not changed, for example, even if the underlying type is the same `[FilePath]`.

But I see the semantic types as being a set that grows over time and as we understand what is needed, and become something that every IDE should support in some way as part of hie integration. And we should be able to fall back to a vanilla semantic type that says something like "raw".

And this possibly has a bearing on #79